### PR TITLE
Create GEMM performance report comparing against rocBLAS

### PR DESCRIPTION
**Pull request by krzysz00**
1. Refactor perf report scripts to allow both convolution and gemm
testing
2. Add gemm testing to Jenkinsfile
3. Fix an issue that make the output precision of tables too high
4. Rename `MIOpenDriver.py` to `perfRunner.py` because it's pretty generic now

### CODE REVIEW

File: mlir/utils/performance/perfRunner.py [hunk 1]
```diff
@@ -360,7 +389,169 @@ def __init__(self, dtype: str, direction: str, layout: str,
...
+                raise ValueError(f"Unknown GEMM config argument {opt} -> {val}")
+        for v in [dtype, g, m, k, n, transA, transB]:
+            if v is None:
+                raise ValueError("Incomplete GEMM configuration")
+
+        return cls(dtype, g, m, k, n, transA, transB, arch)
+
+    def __init__(self, dtype: str, g: int, m: int, k: int, n: int,
+                 transA: bool, transB: bool, arch: str):
+        if dtype not in {"f16", "f32", "bf16", "i8"}:
```
Comment (zhanglx13) [inline mlir/utils/performance/perfRunner.py:513]: So are we testing bf16 or not given 'DATA_TYPES_GEMM' does not include bf16.

File: mlir/utils/performance/perfRunner.py [hunk 2]
```diff
@@ -517,19 +680,27 @@ def getArch():
...
-    python3 MIOpenDriver.py -b
-    python3 MIOpenDriver.py --batch_miopen
-    python3 MIOpenDriver.py -- conv -F 1 -f NCHW -I NCHW -O NCHW -n 256 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
-    python3 MIOpenDriver.py --miopen -- conv -F 1 -f NCHW -I NCHW -O NCHW -n 256 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
-    python3 MIOpenDriver.py --miopen_use_tuned_mlir
-    python3 MIOpenDriver.py --miopen_use_untuned_mlir
+    python3 perfRunner.py
+    python3 perfRunner.py --batch_both -o=output_file.csv
+    python3 perfRunner.py -b
+    python3 perfRunner.py --batch_miopen
```
Comment (zhanglx13) [inline mlir/utils/performance/perfRunner.py]: This option does not exist anymore.
Also, could you please add more examples, including gemm perf runs.

**Review by zhanglx13 — APPROVED** [review]
LGTM.

File: mlir/utils/jenkins/Jenkinsfile [hunk 3]
```diff
@@ -111,13 +111,20 @@ void postProcessPerfRes(String chip) {
         alwaysLinkToLastBuild: false,
         keepAll: true,
         reportDir: 'build',
-        reportFiles: "${chip}_MLIR_Performance_Changes.html,${chip}_MLIR_vs_MIOpen.html",
+        reportFiles: "${chip}_MLIR_Performance_Changes.html,${chip}_MLIR_vs_MIOpen.html","${chip}_MLIR_Performance_Changes_Gemm.html","${chip}_MLIR_vs_rocBLAS.html"
```
Comment (zhanglx13) [inline mlir/utils/jenkins/Jenkinsfile]: Jenkins is complaining the syntax here. Maybe you can put the file in the same quotes.

File: mlir/utils/performance/perfRunner.py (hunk 1 above)
Comment (krzysz00) [reply mlir/utils/performance/perfRunner.py:513]: We could test bf16 but rocBLAS didn't support it on Navi so I took it out of the list

**Comment by krzysz00** [comment]
PR has been shown to work at http://rocmhead.amd.com:8080/blue/organizations/jenkins/MLIR%2Fmlir/detail/PR-830/15/pipeline, will be cheating it in

**Comment by krzysz00** [comment]
Update: looks like we've got a real CI pass so I'll take that instead

**Pull request merged**